index.html

```diff
@@ -1644,6 +1644,72 @@ <h2>
     </section>
   </section>
   </section>
+
+  <section>
+    <h2>Verification Algorithm</h2>
+    <p>
+This specification might be used with many different key discovery protocols.
+As such, discovery of verification keys is described in a different section of this document, 
+and is assumed to have succeeded prior to beginning the verification process.
+    </p>
+    <p>
+As a general rule, verifiers SHOULD strive to minimize the processing of untrusted data.
+This includes minimizing any processing of the protected header, unprotected header, or payload as part of the key discovery procedures.
+    </p>
+
+    <p>
+When verifying a credential or presentation secured with SD-JWT, the algorithm defined in 
+<a href="https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt-06#name-verification-of-the-sd-jwt">
+  Verification of the SD-JWT
+</a> MUST be followed.
+    </p>
+
+    <p>
+When verifying a credential or presentation secured with COSE Sign 1, the algorithm defined in 
+<a data-cite="RFC9052#section-4.4">
+Signing and Verification Process
+</a> MUST be followed.
+    </p>
+
+    <p>
+After verification has succeeded, additional validation checks SHOULD be performed.
+    </p>
```

Comment:
Can you format this and the validation algorithm according to the conventions in https://infra.spec.whatwg.org/#algorithms (whether or not you actually cite Infra)? I'd like to see something like

```markdown
To <dfn>verify a JOSE/COSE verifiable credential or presentation</dfn>, given a [=MIME type=] |mediaType| and a [=byte sequence=] |input|, the verifier MUST:
1. If |mediaType|'s [=MIME type/essence=] is "application/vc...+sd-jwt":
    1. Run the algorithm in <a data-cite="draft-ietf-oauth-selective-disclosure-jwt-06#name-verification-of-the-sd-jwt">Verification of the SD-JWT</a> with the following modifications:
       * The verifier MUST NOT dereference the issuer or key during this process.
       * Any particular claims to look for in the "Check that the SD-JWT is valid using claims such as nbf, iat, and exp" step.
       * Anything else from SD-JWT that asks for the application to customize it, and which can be determined at this level.
1. If |mediaType|'s [=MIME type/essence=] is "application/vc...+cose": ...
```

With any mistakes I've made in that sketch fixed up, of course.

Reply:
Yes, although we probably don't want to repeat algorithm definitions if we can instead cite them directly in the IETF text.

I am hopeful we can do a bit of both here, use the infra style algorithm definitions and cause the reader to follow the steps as defined in the normative references without repeating normative algorithm definitions.